services/core-api/app/api/now_applications/resources/now_application_progress_resource.py

```diff
@@ -75,4 +82,17 @@ def put(self, application_guid, application_progress_status_code):
         existing_now_progress.end_date = data['end_date']
         existing_now_progress.save()
 
+        # update application status if referral/consultation/Public Comment are all complete.
```

Comment:
I think the logic for determining if an application has had each of REF, CON, and PUB completed is more complex. We need to grab the **most recent** progress objects for REF, CON, and PUB. If any of those is null, then we know referral is not completed. If the end date for each of those most recent records exists, then we know that referral has completed.

Reply:
The below code is grabbing all the end_dates for any record of type REF/CON/PUB. 
The array could have 1-3 entries, if end_dates exist, it will update the status, if any of the end_dates are None it will not change anything. 
so if all three are in progress, the array will be [None, None, None] - Status will not be updated.

I can walk you through a demo.

Reply:
But what if there were only progress records for PUB, for example, it would still think referral was complete if PUB had an end date but REF and CON had no records yet.

Reply:
yes - that's what the ticket wanted. 
REF/CON/PUB is not required, if one is started it must be completed, but there is no logic to force them to all be started.

This is what the ticket stated - I will confirm with Brian/Ashley
"When all of Consulation, Referral or PC that have been started and are now complete. Anything not started does not need to be considered. If something gets started after another is complete, reset status to Referred"

Reply:
Confirmed with Ashley - This is the workflow they would like for right now.